controllers/etcd/reconciler.go

```diff
@@ -427,12 +390,13 @@ func (r *Reconciler) reconcileEtcd(ctx context.Context, logger logr.Logger, etcd
 		return reconcileResult{err: err}
 	}
 
-	values, err := r.getMapFromEtcd(etcd, r.config.DisableEtcdServiceAccountAutomount)
+	saDeployer := componentserviceaccount.New(r.Client, componentserviceaccount.GenerateValues(etcd, r.config.DisableEtcdServiceAccountAutomount))
+	err = saDeployer.Deploy(ctx)
 	if err != nil {
 		return reconcileResult{err: err}
 	}
 
-	err = r.reconcileServiceAccount(ctx, logger, etcd, values)
+	values, err := r.getMapFromEtcd(etcd, r.config.DisableEtcdServiceAccountAutomount)
```

Comment:
I think in other PRs you will also remove the need for computing the values for roles and rolebinding. For the time being can you please rename this variable to depict that these are values to be used for roles and rolebinding?

Reply:
Renamed.